docs/developer-docs/web-apps/verifiable-credentials/how-it-works.mdx

```diff
@@ -0,0 +1,84 @@
+keywords: [intermediate, concept, identity, credentials, verifiable credentials]
+---
+
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+import useBaseUrl from "@docusaurus/useBaseUrl";
+
+# How it works
+
+<MarkdownChipRow labels={["Intermediate", "Concept"]} />
+
+## Overview
+
+Verifiable credentials (VCs) on ICP rely on Internet Identity (II), which provides different and unlinkable identities for each dapp. A dapp utilizing verifiable credentials can’t ask for a user's credentials from the issuer of the verifiable credential directly. When a relying party wants to obtain a credential from an issuer, it must request it through an identity provider such as Internet Identity.
+
+Internet Identity acts as a communication link between the relying party and issuer API, connecting with the issuer API and asking for the credentials for the user. After receiving the credentials, they are forwarded to the relying party.
+
+<div class="text--center">
+  <img
+    src={useBaseUrl("/img/docs/vcs-1.png")}
+    alt="Verifiable Credential Workflow"
+    width="800"
+  />
+</div>
+
+## User's perspective
+
+From the user's perspective, the relying party opens a new window to allow the credential, which is closed when the flow finishes.
+
+<div class="text--center">
+  <img
+    src={useBaseUrl("/img/docs/vcs-2.png")}
+    alt="Verifiable Credential Workflow"
+    width="800"
+  />
+</div>
+
+## Developer's perspective
+
+From the developer’s perspective, the interaction between dapps uses window messages and calls to canisters:
+
+1. The user accesses the relying party and starts the process to request credentials. This may be an action such as logging into a dapp or accessing a private service that requires authorization.
+
+2. The relying party starts the flow with Internet Identity (II):
+
+    a. The relying party opens a new II window and waits for a window message.
+
+    b. The frontend of II sends a window message to start the request process.
+
+    c. The relying party replies with the credentials request and issuer data for the request.
+
+3. The II frontend calls the issuer canister to retrieve the consent message.
+
+4. The II frontend shows all the details to the user: the relying party, the consent message, and the issuer URL, then waits for the user action.
+
+5. The user decides whether to allow or revoke the action.
+
+    a. If allowed, the II frontend requests the ID alias to the II backend.
+
+6. The II backend calculates an ID alias which is an opaque identifier.
+
+7. The II frontend requests the credential from the issuer API.
+
+8. The issuer receives the request for the credential, validates the call from II, and replies with a verifiable presentation if the user fulfills the credential's requirements.
+
+9. The II frontend receives the credential from the issuer and forwards it to the relying party, closing the window.
+
+10. The relying party verifies the credentials.
+
+<div class="text--center">
+  <img
```

Comment:
It would be good if the terms in the diagram and text matched 100%. Now, some terms in the diagram are not mentioned in the text. "Attribute request" and "Cert. attribute" for instance. 

I also have a sense the diagram would be easier to follow as a sequence diagram. But, perhaps that is just a matter of personal taste.

Reply:
Yes, we are working on new diagrams. The current ones are temporary to give an idea of what we want to show in the new ones.